config/base/derive/src/proxy.rs

```diff
@@ -128,9 +132,12 @@ pub fn impl_load_from_disk(ast: &StructWithFields) -> TokenStream {
         impl #disk_trait for #proxy_name {
             type Error = #error_ty;
             fn from_path<P: AsRef<std::path::Path> + std::fmt::Debug + Clone>(path: P) -> Result<Self, Self::Error> {
-                let file = std::fs::File::open(path)?;
-                let reader = std::io::BufReader::new(file);
-                serde_json::from_reader(reader)?
+                let mut file = std::fs::File::open(path)?;
+                // String has better parsing speed, see [issue](https://github.com/serde-rs/json/issues/160#issuecomment-253446892)
```

Comment:
This is a micro-optimisation, and I'd normally be against it, however, in this case we also have a few benefits, namely that the file can be close very quickly, or that we can use another method of obtaining the config later on.